tests/specs/lemmas.k

```diff
@@ -325,11 +336,38 @@ module LEMMAS-HASKELL [symbolic, kore]
     rule #range(M [ N := BUF:ByteArray ], L, K) => BUF [ L -Int N .. minInt(K, #sizeByteArray(BUF) -Int (L -Int N)) ] ++ #range(M, N +Int #sizeByteArray(BUF), K -Int minInt(K, #sizeByteArray(BUF) -Int (L -Int N)))  requires K  >Int 0 andBool L <Int N +Int #sizeByteArray(BUF) andBool L >=Int N                          [simplification]
     rule #range(M [ N := BUF:ByteArray ], L, K) => #range(M, L, K)                                                                                                                                                     requires K  >Int 0                                           andBool L >=Int N +Int #sizeByteArray(BUF) [simplification]
 
+
+    rule #range(M [ N := BUF:ByteArray ], N, K ) => 
+      BUF ++ #range( M, N +Int minInt(K, #sizeByteArray(BUF)), K -Int minInt (K, #sizeByteArray(BUF)))  
+      requires K >Int 0 [simplification]
+
+
     rule 0 <=Int lengthBytes ( _ ) => true [smt-lemma, simplification]
 
     rule BA:ByteArray   ==K #buf(32, DATA) => #buf(32, DATA) ==K              BA                                       [simplification, concrete(BA)]
     rule #buf(32, DATA) ==K BA:ByteArray   =>          DATA  ==Int #asInteger(BA) requires #sizeByteArray(BA) <=Int 32 [simplification, concrete(BA)]
 
+    rule #range( M [ N1 := BUF1:ByteArray ] [ N2 := BUF2:ByteArray ], L, K ) =>
+         #range( M                          [ N2 := BUF2           ], L, K )
+      requires N1 +Int #sizeByteArray(BUF1) <=Int L [simplification]
+
+    rule #range (M, N1 +Int N2, K ) => #range( #range( M, N2, #sizeByteArray(M) -Int N2 ), N1, K ) 
+      [simplification, concrete(N2), concrete(M)]
```

Comment:
That's a weird looking lemma, but makes sense to simplify the concrete portion as much as possible.

Reply:
Ah actually, maybe a conuterexample could be had with `N2` negative?

Suppose `M ==K .ByteArray`, `N1 ==Int 3`, `N2 ==Int -5`, `K ==Int 2`.

Reduction without this lemma:

```
#range(.ByteArray, 3 +Int -5, 2) =>
.ByteArray [ -2 .. 2 ] =>
.ByteArray
```

With this lemma:

```
#range(.ByteArray, 3 +Int -5, 2) =>
#range(#range(.ByteArray, -5, 0 -Int -5), 3, 2) =>
(.ByteArray [ -5 .. -5 ]) [ 3 .. 2 ] =>
.ByteArray [ 3 .. 2 ] =>
b"\x00\x00"
```

Reply:
I think the only changes needed are that `N1` and `N2` are positive. You could also probably require `K` positive for simplicity of reasoning.